src/vis/Vis.tsx

```diff
@@ -113,6 +116,12 @@ export function Vis({
         },
   );
 
+  const setExternalConfigRef = useSyncedRef(setExternalConfig);
+  useEffect(() => {
+    setExternalConfigRef.current?.(visConfig);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [JSON.stringify(visConfig), setExternalConfigRef]);
```

Comment:
@puehringer Can you please have a look why `JSON.stringify(visConfig)` is necessary here? Removing this will cause an infinite loop, according to Zach's initial post above.

Reply:
@thinkh Well the `JSON.stringify()` makes it comparable while ignoring references, i.e. `{} !== {}` but `'{}' === '{}'`. Therefore, somewhere in your code you are not changing the "value"/string representation, but still assign a new object to the visConfig. You could also do a deepEqual check *before* using the setter to avoid a rerender completely.

Reply:
To follow up on this after talking to Michael, the problem here was not actually in the vis component, but an issue of not memoizing the props being passed to the vis component in ordino 2.0. Changes were made to the https://github.com/Caleydo/ordino/pull/378 PR to account for this. The stringify was also removed.

Reply:
Thanks for the explanation.